### ISSUE TITLE
RouteOptions.without(waypoint:) doesn't return filtered waypoints.

### ISSUE THREAD

**Issue by MaximAlien** [mapbox/mapbox-navigation-ios#2472]
**Mapbox Navigation SDK version:** `1.0.0-beta1`

`RouteOptions.without(waypoint:)` doesn't return filtered waypoints, functions always returns the same list of waypoints. Consider example where waypoint which is set as without is: `Dropped Pin #1`.
**_Actual_**:
```
Before filter: [User location, Dropped Pin #1, Dropped Pin #2]
After filter: [User location, Dropped Pin #1, Dropped Pin #2]
```

**_Expected:_**
```
Before filter: [User location, Dropped Pin #1, Dropped Pin #2]
After filter: [User location, Dropped Pin #2]
```

**Comment by MaximAlien** [comment]
This is regression issue, which is not related to Navigation SDK. I've checked `0.37.1` version of Navigation SDK and waypoint removal works as expected. In short, `- (CLLocationCoordinate2D)convertPoint:(CGPoint)point toCoordinateFromView:(nullable UIView *)view;` now returns waypoint coordinate which slightly differs from the one which was originally added, it prevents us from performing exact comparison, which in turns leads to incorrect filtering results.

Here're some comparison examples:
**0.37.1:**
```
Waypoint which is returned from `convert(_:toPointTo:)`:
(lldb) po waypoint.coordinate
▿ CLLocationCoordinate2D
  - latitude : 37.3388998652851
  - longitude : -122.03229148359603

Original coordinates:
(lldb) po waypoints[0].coordinate
▿ CLLocationCoordinate2D
  - latitude : 37.33233141
  - longitude : -122.0312186

(lldb) po waypoints[1].coordinate
▿ CLLocationCoordinate2D
  - latitude : 37.3388998652851
  - longitude : -122.03229148359603

(lldb) po waypoints[2].coordinate
▿ CLLocationCoordinate2D
  - latitude : 37.34942223422739
  - longitude : -122.0323218641307
```

**1.0.0-beta.1:**
```
Waypoint which is returned from `convert(_:toPointTo:)`:
(lldb) po waypoint.coordinate
▿ CLLocationCoordinate2D
  - latitude : 37.34006
  - longitude : -122.032226

Original coordinates:
(lldb) po waypoints[0].coordinate
▿ CLLocationCoordinate2D
  - latitude : 37.33233141
  - longitude : -122.0312186

(lldb) po waypoints[1].coordinate
▿ CLLocationCoordinate2D
  - latitude : 37.340059948352376
  - longitude : -122.03216273756487

(lldb) po waypoints[2].coordinate
▿ CLLocationCoordinate2D
  - latitude : 37.35052987404731
  - longitude : -122.03230001349077
```

As you can see in second case coordinate returned by `Mapbox.framework` is slightly different. This means that `waypoints[1].coordinate` will not be removed. I think I'll create ticket on `mapbox-gl-native-ios` side to take a look at this.

**Comment by MaximAlien** [comment]
I've revisited current implementation and added fix in Navigation SDK Example. Several API changes were introduced in Mapbox Directions since `v0.30.0` Navigation SDK release and in one of those changes `Route` class dropped usage of `RouteOptions`, this eventually led to this defect, because Example app kept storing original waypoints.

Fix reassigns waypoints which were placed by user each time with waypoints which were returned by Mapbox Directions. Those coordinates might be slightly different from original ones, since they are snapped to road.

At the same time I've found inconsistency in Maps SDK with conversion: coordinate -> point -> coordinate. Original and converted coordinates are not equal after such conversion. Such inconsistency doesn't cause any issues in Navigation SDK and this point, but probably worth taking a look. Unit test which reproduces it can be found [here](https://github.com/mapbox/mapbox-gl-native-ios/blob/5971a71183e0f9c6b02a0cce3721f7c0f3854ecb/platform/darwin/test/MGLMapViewTests.m#L155-L166).
